src/components/table/Table.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { FC, ReactElement } from "react";
+import { TableWrapper, Row, Body, HeadRow, Cell, Title } from "./Table.styled";
+
+export interface TableProps {
+  headerCells: ICell[];
+  rows: Row[];
+  // Optional Title component
+  title?: string | ReactElement;
+}
+
+export type CellSize = "xs" | "sm" | "md" | "lg";
+
+export interface ICell {
+  // if undefined, defaults to "sm"
+  size?: CellSize;
+  value: string | ReactElement;
+}
+
+export interface Row {
+  cells: ICell[];
+}
+
+const Table: FC<TableProps> = ({ headerCells, rows, title }) => {
+  return (
+    <TableWrapper>
+      {title && typeof title === "string" ? (
+        <Title>{title}</Title>
+      ) : title ? (
+        title
+      ) : null}
+      <HeadRow>
+        {headerCells.map((cell, index) => {
+          return (
+            <Cell key={index} size={cell.size}>
+              {cell.value}
+            </Cell>
+          );
+        })}
+      </HeadRow>
+      <Body>
+        {rows.map((row, ridx) => {
+          return (
+            <Row key={ridx}>
```

Comment:
I think this is one example where consumers might need more control, using the array index as key only works when not rearranging lists, so if a consumer wants to do something on the cells (say sort or move them), this might break the table for them. 
There is not a better choice we can make ourselves though as there is no natural key for a cell of a generic table, so the only way is to leave the choice to consumers.

Reply:
good call, you probably need a way to optionally specify an id for each cell.  if you use a primitive value, you can probably just use the value itself, but it gets more complex if its a react component being rendered

Reply:
we might need to just test this to see what happens if we sort the rows and pass the props back down though, it might just work since all the props are different and force a rerender, im not sure